### PR TITLE
add doc for hyperparameter search

### PR DESCRIPTION
**Pull request by sywangyi**
# What does this PR do?

Fixes # (issue)
add doc for hpo

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

Documentation: @sgugger

### CODE REVIEW

**Comment by sywangyi** [comment]
@yao-matrix

File: docs/source/en/HPO_train.mdx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+```py
+>>> trainer = Trainer(
+...     model=model,
+...     args=training_args,
+...     train_dataset=small_train_dataset,
+...     eval_dataset=small_eval_dataset,
+...     compute_metrics=compute_metrics,
+...     tokenizer=tokenizer,
+...     model_init=model_init,
+...     data_collator=data_collator,
```
Comment (sgugger) [inline docs/source/en/HPO_train.mdx:101]: You should only pass the `model_init`, not the `model`.

**Review by sgugger — COMMENTED** [review]
Thanks for the PR. I left a few comments.
Also can we lowercase the name of the added file to `hpo_train.mdx`? This is for consistency with the rest of the doc.

File: docs/source/en/HPO_train.mdx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+...     direction="maximize",
+...     backend="optuna",
+...     hp_space=optuna_hp_space,
+...     n_trials=20,
+...     compute_objective=compute_objective,
+... )
+```
+
+## Hyperparameter search For DDP refinetune
+Currently, Hyperparameter search for DDP is enabled for optuna and sigopt. Only rank0 will generate the search trial and pass the argument to other ranks.
```
[the remark this answers is not in the document]
Comment (sywangyi) [reply docs/source/en/HPO_train.mdx]: thanks for the review, update the doc accordingly

**Review by sgugger — APPROVED** [review]
Thanks!

**Pull request merged**